File: smart_travel_FE/lib/data/models/banner/banner_update_request.dart
class BannerUpdateRequest {
  final int id;
  final String title;
  final String imageUrl;
  final String? linkUrl;
  final String? description;
  final bool active;

  BannerUpdateRequest({
    required this.id,
    required this.title,
    required this.imageUrl,
    this.linkUrl,
    this.description,
    required this.active,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'imageUrl': imageUrl,
      'linkUrl': linkUrl,
      'description': description,
      'active': active,
    };
  }
}

File: smart_travel_BE/src/main/java/com/example/smart_travel_BE/dto/banner/request/BannerUpdateRequest.java
package com.example.smart_travel_BE.dto.banner.request;

import jakarta.validation.constraints.NotBlank;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class BannerUpdateRequest {
    @NotBlank(message = "BANNER_NAME_BLANK")
    String title;

    @NotBlank(message = "IMAGE_URL_BLANK")
    String imageUrl;

    String linkUrl;
    String description;
    boolean active;
}